cogs/utils/paginator.py

```diff
@@ -0,0 +1,268 @@
+import asyncio
+import discord
+from discord.ext.commands import Paginator as CommandPaginator
+from discord.ext.commands import Context
+import typing as t
+
+FIRST_EMOJI = "\u23EE"  # [:track_previous:]
+LEFT_EMOJI = "\u2B05"  # [:arrow_left:]
+RIGHT_EMOJI = "\u27A1"  # [:arrow_right:]
+LAST_EMOJI = "\u23ED"  # [:track_next:]
+PAGE_NUM_EMOJI = "\N{INPUT SYMBOL FOR NUMBERS}"
+STOP_EMOJI = "\N{BLACK SQUARE FOR STOP}"
+INFO_EMOJI = "\N{INFORMATION SOURCE}"
+
+
+class CannotPaginate(Exception):
+    pass
+
+
+class Pages:
+    """Implements a paginator that queries the user for the pagination interface."""
+
+    def __init__(self, ctx: Context, entries: t.List[str], per_page: int = 12, show_entry_count: bool = True) -> None:
+        self.bot = ctx.bot
+        self.entries = entries
+        self.message = ctx.message
+        self.channel = ctx.channel
+        self.author = ctx.author
+        self.per_page = per_page
+        pages, left_over = divmod(len(self.entries), self.per_page)
+        if left_over:
+            pages += 1
+        self.maximum_pages = pages
+        self.embed = discord.Embed(colour=discord.Colour.blurple())
+        self.paginating = len(entries) > per_page
+        self.show_entry_count = show_entry_count
+        self.reaction_emojis = [
+            (FIRST_EMOJI, self.first_page),
+            (LEFT_EMOJI, self.previous_page),
+            (RIGHT_EMOJI, self.next_page),
+            (LAST_EMOJI, self.last_page),
+            (PAGE_NUM_EMOJI, self.numbered_page),
+            (STOP_EMOJI, self.stop_pages),
+            (INFO_EMOJI, self.show_help),
+        ]
+
+        if ctx.guild is not None:
+            self.permissions = self.channel.permissions_for(ctx.guild.me)
+        else:
+            self.permissions = self.channel.permissions_for(ctx.bot.user)
+
+        if not self.permissions.send_messages:
+            raise CannotPaginate("Bot cannot send messages.")
+
+        if not self.permissions.embed_links:
+            raise CannotPaginate("Bot does not have embed links permission.")
+
+        if self.paginating:
+            # verify we can actually use the pagination session
+            if not self.permissions.add_reactions:
+                raise CannotPaginate("Bot does not have add reactions permission.")
+
+            if not self.permissions.read_message_history:
+                raise CannotPaginate("Bot does not have Read Message History permission.")
+
+    def get_page(self, page: int) -> t.List[str]:
+        base = (page - 1) * self.per_page
+        return self.entries[base : base + self.per_page]
```

Comment:
```suggestion
        base = (page - 1) * self.per_page
        stop = base + self.per_page
        return self.entries[base:stop]
```
This way `flake8` and `black` don't conflict over whitespaces before `:`